stdlib/public/Concurrency/Actor.cpp

```diff
@@ -1254,16 +1269,23 @@ void DefaultActorImpl::giveUpThread(RunningJobInfo runner) {
                                     firstPreprocessed,
                                     firstNewJob,
                                     overridesToWake);
+      traceJobQueue(this, firstNewJob);
 
       // Try again.
       continue;
     }
 
-#define LOG_STATE_TRANSITION                                                   \
-  SWIFT_TASK_DEBUG_LOG("actor %p transitioned from %zx to %zx (%s)\n", this,   \
-                       oldState.Flags.getOpaqueValue(),                        \
-                       newState.Flags.getOpaqueValue(), __FUNCTION__)
-    LOG_STATE_TRANSITION;
+#define ACTOR_STATE_TRANSITION                                                 \
+  do {                                                                         \
+    SWIFT_TASK_DEBUG_LOG("actor %p transitioned from %zx to %zx (%s)\n", this, \
+                         oldState.Flags.getOpaqueValue(),                      \
+                         newState.Flags.getOpaqueValue(), __FUNCTION__);       \
+    concurrency::trace::actor_state_changed(                                   \
+        this, newState.FirstJob.getRawJob(),                                   \
+        newState.FirstJob.needsPreprocessing(),                                \
+        newState.FirstJob.isOverride(), newState.Flags.getOpaqueValue());      \
```

Comment:
Is there a goal that tracing makes any sense to anyone without knowledge of the current runtime implementation?

Reply:
Not at this level. The goal is to get the info out in a compact and fast manner, and then interpreting it into something more sensible for a human reader is up to whatever receives it.